contracts/ContractRegistry.sol

```diff
@@ -1,32 +1,44 @@
-pragma solidity ^0.5.11;
+pragma solidity 0.5.11;
 
-import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/access/Roles.sol";
 
 
-contract ContractRegistry is Ownable {
-    mapping (string => address) private contractAddresses;
+contract ContractRegistry {
+  using Roles for Roles.Role;
 
-    event UpdateContract(string name, address indexed contractAddress);
+  Roles.Role private admin;
 
-    function updateContractAddress(
-        string calldata _name,
-        address _address
-    )
-        external
-        onlyOwner
-        returns (address)
-    {
-        contractAddresses[_name] = _address;
-        emit UpdateContract(_name, _address);
+  mapping (string => address) private contractAddresses;
 
-        return _address;
-    }
+  event UpdateContract(string name, address indexed contractAddress);
 
-    function getContractAddress(string calldata _name)
-        external
-        view
-        returns (address)
-    {
-        return contractAddresses[_name];
-    }
+  constructor()
+    public
+  {
+    admin.add(msg.sender);
+  }
+
+  function updateContractAddress(
+    string calldata _name,
+    address _address
+  )
+    external
+    returns (address)
+  {
+    require(bytes(_name).length != 0, "name must be at least one character long");
+    require(admin.has(msg.sender), "Admin role required");
```

Comment:
`require(admin.has(msg.sender), "Admin role required");` have you done it this way because you don't want to define a modifier for just one use or is there another reason ?

Reply:
Yeah, you're spot on, I know this is gonna be a super simple contract and didn't want to define a reusable modifier for one line! :fist_raised:

Reply:
That sounds good to me, thank you for clarifying 💎